---
layout: page
permalink: /travelling/milan-italy-2017
title: COMBINE 2017, Milan, Italy
---

This year in July I participated a [COMBINE workshop](http://co.mbine.org/events/COMBINE_2017) held by [Computational Modeling in Biology Network](http://co.mbine.org/) in [Università degli Studi di Milano Bicocca](http://www.unimib.it/go/102/Home/English), Milan, Italy. I had a [presentation](/files/presentations/combine2017.pdf) about _Biochemical Space: A framework for formal description and annotation of complex biological processes_.

The community behind this conference focuses on developing standards and formats in systems biology and related fields. Therefore, it was quite a challenge to go there with our work, which does not really meet the standards.

![biccoca](/images/italy-2017/biccoca.jpeg)

Sunday (8.10.2017)
------------------

I left Brno early morning to catch flight from Prague at 1pm. Since I'm quite a lucky person when it comes to traveling, everything went good. The flight was little bit delayed (20 minutes), but that was the only issue.

In Milan Airport Malpansa, I wasn't really sure how to get into the city. But following instructions by Google Maps was easy and since they are really accurate, there were no troubles once again. Even buying tickets in not-really-reliable-looking ticket machines was surprisingly easy (yes, they had English there!). So I bought a random ticket and it was really the one I needed (at this point you probably realised I own an infinite bag of luckiness) for the train to Milan center. Journey in air-conditioned, almost empty and brand new train was enjoyable. It took about 45 minutes. Then I had to change to Milan subway (Metropolitana di Milano) No. 5 to get to my hotel. There was a ticket machine, this time it was quite clear which ticket I need (of course the standard one since I go on a conference about standards). This took just about 10 minutes and then I got off. After 5 minutes of walking I reached my hotel (Hotel degli Arcimboldi). When saw its narrow shape I just thought “I hope I don't have room in it's narrowest corner”.

![hotel](/images/italy-2017/IMG_20171008_183654.jpg)

Well, of course I had (there we go with the infinite luck). But fortunately it just looked bad but wasn't. Room was nice and big enough.

My first day ended with shopping nearby (I had no water and food) and struggling with connecting to the Internet. I succeeded with all these challenging tasks and spent rest of the evening by relaxing. An interesting fact was digital display price tags in the shop.

![price-tags](/images/italy-2017/IMG_20171008_181400.jpg)

It might be quite investment, but from a long-range perspective it is an excellent idea. I want to mention (since I haven't been to Italy before) that the inland of the country looks very similar to Czech republic or Slovakia (just slightly warmer). Just a typical European country.

Monday (9.10.2017)
------------------

The key word of the first day of COMBINE workshop was in_vited speaker_. The entire day was dedicated to invited speakers. Quite unusual, but practical - people who are interested only in those talks might leave after the first day. In the following goes short summary of most interesting talks and their most relevant messages for our lab resulting from the talks:

#### James Faeder

**Facts:** James is one of heading persons in rule-based modelling in systems biology. He is actually the author of [BioNetGen Language](http://bionetgen.org/index.php/Main_Page), which was for example used in my bachelor thesis.

**Title:** _Toward integrated models of cell behavior: How rule-based modeling can contribute_

James has given a nice introduction to rule-based modelling, motivated its usage, and explained what is it good for. From the beginning I could feel the purpose of this community is just creating standards but is less interested in its further usage. In the talk were highlighted descriptive properties of the rule-based languages and approaches to simulate them, but there was no mention about its potential in for example static analysis.

![james-feader](/images/italy-2017/IMG_20171009_104145.jpg)

One note to simulation part of rule-based languages relevant for language developed in our language: Biochemical Space Language currently uses indirect simulation approach (enumerating reactions first and from them build ODEs model), which is generally the fastest approach. But the fact is that problem might occur when a model is too large - enumerating all the species and reactions might be impossible or too memory consuming.

Such a huge models were motivation for BNGL to use NFsim simulation engine, which uses direct Monte-Carlo-like approach. James showed suspicious graph where he compared few other approaches and showed using NFsim is the best one. This slide immediately made several people complaining about the graph, because it wasn't really complete.

Later, after the talk, I was involved in a discussion, where we were talking about it. I pointed to general problem with NFsim - there is one expensive step in the algorithm - application of the rule. This might not be that bad if number of different species in model is low. That's actually the information which was missing in James's figure. As the number of species rises, the mentioned step becomes more and more expensive and the approach becomes generally less efficient.

Next he presented some methods for visualising such systems, where the most interesting ones were extended contact maps. He also mentioned John Sekar's work (his student) which I saw at [SASB 2017](sas-b-new-york-usa-2017.html).

#### Martin Golebiewski

**Title:** _Don’t be lost in the jungle of standards: Cataloguing, harmonizing and application of modelling standards_

Martin gave a talk which was an overview to all standards and related tools in systems biology (at least those somehow involved in COMBINE organisation). From his talk I would like to highlight [The NormSys Registry for Modeling Standards](https://normsys.h-its.org/) which aims at surveying standard formats for computational modeling in biology. Also, it is trying to compare their features and applicability. Moreover, he gave nice example why we need standards by pointing to famous Great [Baltimore fire in 1904](https://en.wikipedia.org/wiki/Great_Baltimore_Fire): ”_… One reason for the fire's long duration involved the lack of national standards in firefighting equipment. …_”.

![modelling-tools](/images/italy-2017/IMG_20171009_115024.jpg)

#### Marco Nobile

**Facts:** a local speaker

**Title:** _Turning uncertainty and lack of quantitative data into dynamic models with fuzzy logic_

Marco was talking about fuzzy logic (the truth values are real numbers from interval \[0,1\]) and a standard developed for purposes of models specified with this logic.

#### Carlos Lopez

**Facts:** Carlos is an author of PySB rule-based language and a co-worker to John Bachman.

**Title:** _Reaction flux and information flow in the COX-2 reaction network_

Carlos was generally talking about advantages of embedding a modelling language to a programming language and then explained how they done it with PySB and Python.

#### Padraig Gleeson

**Title:** _The Open Source Brain Initiative, enabling collaborative model development in computational neuroscience_

Padraig was talking about [Open Source Brain project](http://www.opensourcebrain.org/) and NeuroML - a official COMBINE standard XML language for neuroscience.

#### Shailesh Appukuttan

Shailesh was criticising model creation and talking about how modellers often neglect validation of the model they created. Consequently, he presented a validation framework for neuroscience oriented models.

Tuesday (10.10.2017)
--------------------

This day was mostly about talks. Again, I choose the most interesting ones and try to provide short summary of them.

#### Rahuman Sheriff

**Title:** _Resources for semantic enrichment and sharing of models in standard formats_

The talk was mostly about [identifiers.org](http://identifiers.org/), a web service which we are already using for annotating models with out data. What I realised here was the fact that we could use identifiers also for other datasets we are already using (such as GO, PDB, etc.).

#### Mihai Gloth

**Title:** _Towards a new infrastructure for BioModels_

Mihai introduced [BioModels](https://www.ebi.ac.uk/biomodels-main/) model repository for storage and simple analysis of models, which are mostly manually curated. Then he was talking about a new version (already released). One of the new features might be possibility to submit supplementary files with the model, which should improve annotation of the models.

#### Jacky Snoep

**Title:** _The JWS Simulation Database: How linking experimental data and mathematical models enables publication of reproducible model-based results_

[JWS Online](http://jjj.mib.ac.uk/) is a Systems Biology tool for the construction, modification and simulation of kinetic models and for the storage of curated models. This work is very similar to our work, but they do not adapt the idea of a common space (in our case BCS) and also the models are general models, not a domain-specific ones.

#### Mirco Tribastone

**Facts:** Collaboration with Luca Cardelli.

**Title:** _Reduction of ordinary differential equations with ERODE_

Very interesting work, in general they apply a reduction on ODEs before analysing the model itself. The principle is very similar to rule-based approach, but done on ODEs instead of reactions. Relevant for us is that they can do it even with rule-based models. The pipe line then looks like this:

rules → reactions → ODEs → reduced ODEs

but the backward relation to original rule based model is unclear.

#### Claudine Chaouiya

**Facts:** We have already met in [Prague](ismb-eccb-prague-czechia-2017.html) this year.

**Title:** _Standardisation effort from the Consortium for Logical Models and Tools (CoLoMoTo)_

Claudine introduced a logical modelling framework for influence and other boolean-based networks. She was also talking about [qual SBML package](http://sbml.org/Documents/Specifications/SBML_Level_3/Packages/qual), which should be a standard for such models (but when I was checking it later, I was not really sure about it), which might be (maybe) relevant for Signaling pathways group in our lab.

#### Fengkai Zhang

**Facts:** We already had a Skype meeting with Fengkai and were discussing SBML Multi package and its relevance for our language.

**Title:** _Release of SBML Multi — A milestone in addressing rule-based modeling interoperability challenges_

Fengkai introduced SBML multi package and briefly explained its purposes. He also gave an intro to rule-based graphical representation (Simmune). Important for us is that he mentioned our BCSL during his talk (twice!) and labeled it as a potential language, which might fit into the standard. From our consequent discussion I found out the package does not support SBGN visualisation, which is something we wanted to use. However, they want to enable it as a future work.

![fengkai-zhang](/images/italy-2017/IMG_20171010_115915.jpg)

#### Andreas Dräger

**Facts:** We have already met in [Prague](ismb-eccb-prague-czechia-2017.html) this year.

**Title:** _The JSBML project: a fully featured Java API for working with systems biological models_

This was a general talk about JSBML - an API to Java for creating and maintaining SBML models in Java.

I have to say this part of the conference was really interesting and all mentioned talks were very educational for me. Unfortunately, I can't say the same about the rest of the day. There were Parallel Breakouts (three at once) and I joined SBML/CellML/Sed-ML: Multicellular/Agent-based modelling breakout chaired by Dirk Drasdo. However, he wasn't present, so the breakout was also enriched with communication over a video conference. You can imagine there very problems with the audio and video. But worst fact was that it was boring. I know, it sounds maybe little bit severe. But I have to say standards are as important as they are boring.

![dirk-drasdo](/images/italy-2017/IMG_20171010_143543.jpg)

For that reason I decided to skip following breakout session and visit the center of the city. I took a subway to the center. There I got off next to the Duomo di Milano, a very nice cathedral. Since it was getting late, I was just walking around in streets full of expensive clothing shops for about 2 hours. Here is a video from this part of the city:

Wednesday (11.10.2017)
----------------------

The day started with session about Synthetic Biology Open Language (SBOL), which is not really a relevant work for us.

The second session of the day was more interesting, especially because I had a talk there. It was about Biochemical Space in context of Comprehensive Modeling Platform. I was quite confident about the talk and everything went all right. The only problem was the audience. As expected, once this work was not really involved in standards, they were not interested. Only few people had a discussion with me in private, but these were not from the community or were more interested in rule-based modelling. So despite the fact I had a good feeling about my talk, it didn't meet much interest. One of the reasons huge concurrency in this field which is actually using the standards.

Anyway, my chair was Andreas Dräger, who did hid job very well and almost succeeded in pronouncing my name correctly. The slides I used for presentation are available here.

![combine-sedml](/images/italy-2017/combine_sedml.jpg)

In the afternoon's breakout sessions, I participated in Simulation Experiment Description Markup Language ([SED-ML](https://sed-ml.github.io/)) wrap-up. It is a standard XML-based format for encoding simulation setups. It is very similar to `.cps` files produced by Copasi before simulation. This might be relevant for our CMP platform, where we want to improve the model module. We could actually store simulation details (simulation datasets together with all simulation parameters in this file) in this files, which would enable exchangeability and reproducibility for multiple simulation engines (there are APIs for R, Python, and many more).

Unfortunately, the rest of the session was about specifying next version of the standard, which generally meant writing and discussing about UML class diagrams.

After the session, I went to the hotel to take a break. Later, I went to the city to see Castello Sforzesco (a castle). It was very nice and I had the opportunity to see it during the day and also the night, because it was just sunset time.

Thursday (12.10.2017)
---------------------

This day, there were two additional invited speakers chaired by Chris Mayers (a CHARME track).

#### Ugur Sezerman

**Title:** _Identification of disease related pathways in complex diseases via System Biology based analysis of omics data_

Ugur was complaining about how medicine works nowadays, he compared it to Dr. House series. Briefly, you see symptoms and decide the treatment according to them. If new symptom appears, you erase “table” of possible diseases and write new ones and keep trying. Then, he was talking about typical bioinformatics approaches which are being used and about personalised health care. He showed a few cases (patients) where sequencing and identifying significantly altered genes caused the very same type of cancer, also they used this data for personalised cure. He finished his talk with these words:

“_If you don't have cancer, don't change your life style - you never know what it might cause!_”

#### Chris Evelo

This was just a general talk about standards, [CHARME](http://www.cost-charme.eu/), and COST.

![wheel](/images/italy-2017/IMG_20171012_103547.jpg)

After break there was session about BioPAX and SBGN.

#### Augustin Luna

**Title:** _Access and Discover Biological Pathway Information from Pathway Commons and BioPAX_

The goal of BioPAX is to centralise all pathway databases (cca. 650) into one format. He was also talking about [Factoid](https://github.com/PathwayCommons/factoid), a project to digitally capture biological data from academic papers, similar to INDRA.

**Title:** _Introduction and Current Status of the Systems Biology Graphical Notation (SBGN)_

His second talk in this session. It was basically introduction to SBGN and it's usage. He explained there are actually three sub-languages: process description, entity relationship, and activity flow.

![sgbn](/images/italy-2017/IMG_20171012_115424.jpg)

#### Adrien Rougny

**Facts:** _Collaborates with Loïc Paulevé._

**Title:** _SBGNLog: a translation of SBGN PD and SBGN AF into first-order logic to reason on molecular networks_

It was about logic programming in relation to SBGN-like networks. This had actually nothing to do with the visualisation aspects of the SBGN. It was connected to SBGN-ML, a immediate format for SBGN networks.

#### Andreas Dräger

**Facts:** He had the very same talk on [ISMB](ismb-eccb-prague-czechia-2017.html) this year.

**Title:** _Visualization and Creation of Biochemical Networks with Escher_

My original idea was that we could use it for visualising pathways in our platform. All you need is to get it into SBGN/SBML format. However, there is no initial layout, it just imports the network and helps you to build it on your own.

### SBML/SED-ML/CellML/NeuroML using data

Goal of the session: how to use different formats as input for the standards. Example - SEDML uses csv, NuML, tsv, json. They defined set of rules the input file should hold.

During the session, after some comment of Fengkai, I got this idea in context of rule-based modelling:

A rule encodes a set of reactions (a known fact for us), BUT a reaction can be encoded by multiple rules. It's quite obvious, but the leads to kind of funny idea: Assuming we know how to get all rules encoding a reaction, do this in a loop - (1) create reactions from rules and (2) create rules from the reactions. What will we obtain?

### Open SBML Editors Meeting: The Future for SBML

It was discussed whether a next version of SBML is needed - looks like not, packages can cover the missing parts (such as agent based modelling).

![sbml](/images/italy-2017/IMG_20171012_161839.jpg)

They suggested to push journal to accept only models encoded in standards.

Next topic was “why standards are not always used” - they concluded maybe its tool developers they need to target, not the modellers/users.

In the end it showed up they were discussing the very same things back in 2011…

### Social dinner

The day ended with social dinner in [Hangar Bicocca Iuta Bistrot](http://www.hangarbicocca.org/en/). It was actually an art museum and first we had a guided tour through the museum.

![towers](/images/italy-2017/IMG_20171012_193146.jpg)

After that, we finally had the dinner. Funny fact was that they were not able to create both vegetarian and non-vegetarian food, so we all had the vegetarian version. Despite this fact it was delicious.

Friday (13.10.2017)
-------------------

The last day's session was again about SBOL, which was interesting, but had no relevant impact for us.

After break, there were final remarks, greetings, and so on mostly by Martin Golebiewski. There was also a general discussion about future of COMBINE (they consider to promote it to a legal entity). There was also invitation to next COMBINE workshop in Boston next year.

![remarks](/images/italy-2017/IMG_20171013_114042.jpg)

After lunch, I went to the airport and safely returned to Brno. There were literally no issues with the traveling.